File: swap_numbers/swap_numbers.js
var fs  = require("fs");

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        line = line.replace(/\r/gm, ""); // For Windows.

        var words = line.split(" "); // Splits the string by space
        var newWords = [];

        for (var i = 0; i < words.length; i++) {
            var word = words[i][words[i].length-1];
            word += words[i].slice(1, words[i].length-1);
            word += words[i][0];
            newWords.push(word);
        }

        console.log(newWords.join(' '));
    }
});

File: swap_numbers/swap_numbers.cpp
#include <iostream>
#include <sstream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

vector<string> &split(const string &s, char delimiter, vector<string> &elements)
{
    stringstream ss(s);
    string item;

    while (getline(ss, item, delimiter)) {
        elements.push_back(item);
    }

    return elements;
}

vector<string> split(const string &s, char delimiter)
{
    vector<string> elements;
    split(s, delimiter, elements);

    return elements;
}

int main(int argc, char *argv[])
{
    ifstream file(argv[1]);
    string line;

    while(getline(file, line))
    {
        string output_line;
        vector<string> words = split(line, ' ');

        for (string word: words)
        {
            output_line += word[word.size() - 1];
            output_line += word.substr(1, word.size() - 2);
            output_line += word[0];
            output_line += ' ';
        }

        cout << output_line << endl;
    }
}
